internal/os/volume/api.go

```diff
@@ -11,12 +11,38 @@ import (
 
 const formatFilesystem = "ntfs"
 
-// VolAPIImplementor - struct for implementing the internal Volume APIs
-type VolAPIImplementor struct{}
+// API exposes the internal volume operations available in the server
+type API interface {
+	ListVolumesOnDisk(diskNumber uint32) (volumeIDs []string, err error)
+	// MountVolume mounts the volume at the requested global staging target path
+	MountVolume(volumeID, targetPath string) error
+	// UnmountVolume gracefully dismounts a volume
+	UnmountVolume(volumeID, targetPath string) error
+	// IsVolumeFormatted checks if a volume is formatted with NTFS
+	IsVolumeFormatted(volumeID string) (bool, error)
+	// FormatVolume formats a volume with the provided file system
+	FormatVolume(volumeID string) error
+	// ResizeVolume performs resizing of the partition and file system for a block based volume
+	ResizeVolume(volumeID string, sizeBytes int64) error
+	// GetVolumeStats gets the volume information
+	GetVolumeStats(volumeID string) (int64, int64, error)
+	// GetDiskNumberFromVolumeID returns the disk number for where the volume is at
+	GetDiskNumberFromVolumeID(volumeID string) (uint32, error)
+	// GetVolumeIDFromTargetPath returns the volume id of a given target path
+	GetVolumeIDFromTargetPath(targetPath string) (string, error)
+	// WriteVolumeCache writes volume cache to disk
+	WriteVolumeCache(volumeID string) error
+}
```

Comment:
match comments with above. but you can do it later once most review is done.

Reply:
done